addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/utils/ByteArrayFileCache.java

```diff
@@ -0,0 +1,236 @@
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a simple file based cache implementation.
+ * <b>NOTE:</b> This class is a duplicate of
+ * https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.openweathermap/src/main/java/org/eclipse/smarthome/binding/openweathermap/internal/utils/ByteArrayFileCache.java
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class ByteArrayFileCache {
+
+    private final Logger logger = LoggerFactory.getLogger(ByteArrayFileCache.class);
+
+    private static final String CACHE_FOLDER_NAME = "cache";
+    public static final char EXTENSION_SEPARATOR = '.';
+
+    protected final File cacheFolder;
+
+    public ByteArrayFileCache(String servicePID) {
+        // TODO track and limit folder size
+        // TODO support user specific folder
+        cacheFolder = new File(new File(new File(ConfigConstants.getUserDataFolder()), CACHE_FOLDER_NAME), servicePID);
+        if (!cacheFolder.exists()) {
+            logger.debug("Creating cache folder '{}'", cacheFolder.getAbsolutePath());
+            cacheFolder.mkdirs();
+        }
+        logger.debug("Using cache folder '{}'", cacheFolder.getAbsolutePath());
+    }
+
+    /**
+     * Adds a file to the cache. If the cache previously contained a file for the key, the old file is replaced by the
+     * new content.
+     *
+     * @param key the key with which the file is to be associated
+     * @param content the content for the file to be associated with the specified key
+     */
+    public void put(String key, byte[] content) {
+        writeFile(getUniqueFile(key), content);
+    }
+
+    /**
+     * Adds a file to the cache.
+     *
+     * @param key the key with which the file is to be associated
+     * @param content the content for the file to be associated with the specified key
+     */
+    public void putIfAbsent(String key, byte[] content) {
+        File fileInCache = getUniqueFile(key);
+        if (fileInCache.exists()) {
+            logger.debug("File '{}' present in cache", fileInCache.getName());
+        } else {
+            writeFile(fileInCache, content);
+        }
+    }
+
+    /**
+     * Adds a file to the cache and returns the content of the file.
+     *
+     * @param key the key with which the file is to be associated
+     * @param content the content for the file to be associated with the specified key
+     * @return the content of the file associated with the given key
+     */
+    public byte[] putIfAbsentAndGet(String key, byte[] content) {
+        putIfAbsent(key, content);
+
+        // return get(key);
```

Comment:
I don't understand this comment

Reply:
That is not a comment. That is a pitty. I removed it.

Reply:
What do you mean? If that is valid informational comment for you it can have its place, maybe just clarify it a little more. Or do you mean it was there as an error? 🙂